### PR TITLE
Fix scaling

### PR DESCRIPTION
**Pull request by lionel-**
* Fixes text scaling
* Adds a vignette on scaling
* Adds a couple of tests, and `open_manual_tests()` to open html files requiring manual inspection
* Adds `textLength` property to text tags to ensure correct dimensions

### CODE REVIEW

File: src/devSVG.cpp [hunk 1]
```diff
@@ -205,8 +205,10 @@ inline void write_style_dbl(SvgStreamPtr stream, const char* attr, double value,
 }
 
 inline void write_style_fontsize(SvgStreamPtr stream, double value, bool first = false) {
-  if(!first)  (*stream) << ' ';
-  (*stream) << "font-size: " << value << "pt;";
+  if(!first) (*stream) << ' ';
+  // Firefox requires that we provide a unit (even tough px is
```
Comment (hadley) [inline src/devSVG.cpp]: sp - though

File: vignettes/scaling.Rmd [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+straightforward but requires some understanding of the `viewBox` SVG
+attribute which is included in all SVGs produced by svglite. This
+property defines the aspect ratio of the plot (as well as a *user
+coordinate system*, see next section).
+
+
+### Fluid Scaling
+
+The viewBox is determined by the `width` and `height` arguments of
+svglite's device functions (with 10x8 the default). Altough those
```
Comment (hadley) [inline vignettes/scaling.Rmd]: `10" x 8"` to be clear about units?

File: vignettes/scaling.Rmd [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+text within the SVG consistent with the text in the surrounding web
+page. That could be useful, for instance, to create a consistent
+appearance in an HTML presentation. Since the user coordinate system
+defined by the viewBox is unitless, we need to map the figure to its
+natural dimensions. This will ensure a correspondence between the
+scale of the figure and that of the web page.
+
+As mentioned above, the natural scale of svglite's figures is in
+points and is determined by the `width` and `height` arguments that
+you supply to the device functions (10x8 being the default). Altough
```
Comment (hadley) [inline vignettes/scaling.Rmd]: sp. Although

File: vignettes/scaling.Rmd [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+those dimensions are specified in inches, the coordinate system is
+scaled in points. Counting 72 points per inch, the default SVG surface
+is thus 720x576 pt. Note that the CSS standard defines 12pt to be
```
Comment (hadley) [inline vignettes/scaling.Rmd]: I think it's more readable to put a space around the "x"

File: vignettes/scaling.Rmd [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Scaling Issues}
+  %\VignetteEngine{knitr::rmarkdown}
+  \usepackage[utf8]{inputenc}
+---
+
+
+## Scaling SVG Outputs
```
Comment (hadley) [inline vignettes/scaling.Rmd]: I prefer sentence case in titles

**Comment by lionel-** [comment]
latex doesn't like the Japanese characters :/

Is it normal that a PDF vignette is built even though a knitr vignette builder is specified in DESCRIPTION?

**Comment by hadley** [comment]
You definitely shouldn't be getting a pdf vignette. Did you figure out what's going wrong?

**Comment by lionel-** [comment]
yes, it was the manual built from the examples for `svglite()`. Now I know that devtools check doesn't generate those.

**Comment by hadley** [comment]
Ok, great. Is this ready to merge?

**Comment by lionel-** [comment]
yup

**Pull request merged**